src/vm/prestub.cpp

```diff
@@ -829,22 +829,40 @@ Stub * CreateInstantiatingILStub(MethodDesc* pTargetMD, void* pHiddenArg)
     CreateInstantiatingILStubTargetSig(pTargetMD, typeContext, &stubSigBuilder);
     
     // 2. Emit the method body
+    unsigned int numArgs = msig.NumFixedArgs();
     if (msig.HasThis())
     {
         // 2.1 Push the thisptr
         pCode->EmitLoadThis();
+        numArgs++;
     }
 
-    // 2.2 Push the hidden context param 
-    // InstantiatingStub
-    pCode->EmitLDC((TADDR)pHiddenArg);
+#if defined(_TARGET_X86_)
+    if (numArgs < NUM_ARGUMENT_REGISTERS)
+    {
+#endif // _TARGET_X86_
+        // 2.2 Push the hidden context param
+        // InstantiatingStub
+        pCode->EmitLDC((TADDR)pHiddenArg);
+#if defined(_TARGET_X86_)
+    }
+#endif // _TARGET_X86_
 
     // 2.3 Push the rest of the arguments
     for (unsigned i = 0; i < msig.NumFixedArgs();i++)
     {
         pCode->EmitLDARG(i);
     }
 
+#if defined(_TARGET_X86_)
+    if (numArgs >= NUM_ARGUMENT_REGISTERS)
+    {
+        // 2.2 Push the hidden context param
```

Comment:
A nit - when I have provided you with this code, I have forgotten to change the 2.2 here to 2.4 and renumber the 2.4 and 2.5 below. Could you please do that? Otherwise LGTM.